pnet_datalink/src/unix_interfaces.rs

```diff
@@ -34,7 +34,7 @@ pub fn interfaces() -> Vec<NetworkInterface> {
 
     let mut ifaces: Vec<NetworkInterface> = Vec::new();
     unsafe {
-        let mut addrs: *mut libc::ifaddrs = mem::uninitialized();
+        let mut addrs: *mut libc::ifaddrs = mem::MaybeUninit::<libc::ifaddrs>::uninit().as_mut_ptr();
         if libc::getifaddrs(&mut addrs) != 0 {
             return ifaces;
         }
```

Comment:
Shouldn't this (lines 37 through 41) be the following? (extra type annotations and turbofishes added for clarity)

```rust
let mut addrs: MaybeUninit<*mut libc::ifaddrs> = mem::MaybeUninit::<*mut libc::ifaddrs>::uninit();
if libc::getifaddrs(addrs.as_mut_ptr()) != 0 {
    return ifaces;
}
let mut addr: *mut libc::ifaddrs = addrs.assume_init();
```